Tella/Data/Networking/Repositories/DropboxRepository.swift

```diff
@@ -0,0 +1,83 @@
+//
+//  DropboxRepository.swift
+//  Tella
+//
+//  Created by gus valbuena on 9/3/24.
+//  Copyright © 2024 HORIZONTAL. All rights reserved.
+//
+
+import Foundation
+import Combine
+import SwiftyDropbox
+
+protocol DropboxRepositoryProtocol {
+    func handleSignIn() async throws
+    func ensureSignedIn() async throws
+    func signOut()
+}
+
+class DropboxRepository: DropboxRepositoryProtocol {
+    private var client: DropboxClient?
+    private var isCancelled = false
+    private let networkMonitor: NetworkMonitor
+    private var subscribers: Set<AnyCancellable> = []
+    private let networkStatusSubject = PassthroughSubject<Bool, Never>()
+    
+    init(networkMonitor: NetworkMonitor = .shared) {
+        self.networkMonitor = networkMonitor
+        setupNetworkMonitor()
+    }
+    
+    private func setupNetworkMonitor() {
+        networkMonitor.connectionDidChange.sink { isConnected in
+            self.networkStatusSubject.send(isConnected)
+            if !isConnected {
```

Comment:
Do we need this `!isConnected` for the future ?

Reply:
I'll remove it for now